src/trace/context.ts

```diff
@@ -183,6 +213,22 @@ export function readTraceFromEvent(event: any): TraceContext | undefined {
   };
 }
 
+export function readTraceFromEvent(event: any): TraceContext | undefined {
+  if (typeof event !== "object") {
+    return;
+  }
+
+  if (event.headers && typeof event.headers === "object") {
+    return readTraceFromHTTPEvent(event);
+  }
+
+  if (event.Records && event.Records[0] && event.Records[0].eventSource === "aws:sqs") {
```

Comment:
Records might be an empty array, or not an array.
```suggestion
  if (Array.isArray(event.Records) && event.Records.length > 0 && event.Records[0].eventSource === "aws:sqs") {
```

Reply:
nit:
I believe the lambda type definitions also define an interface for common event types. I think SQSEvent is one of them. So you could turn this into a typeguard,

```
import { SQSEvent } from "aws-lambda";
function isSQSEvent(event: any): event is SQSEvent {
   return Array.isArray(event.Records) && event.Records.length > 0 && event.Records[0].eventSource === "aws:sqs";
}
```
Then readTraceFromSQSEvent could accept a SQSEvent, and you don't have to do as much typechecking.

Reply:
Good call!